registry/client/repository.go

```diff
@@ -205,11 +205,30 @@ type tags struct {
 	name    reference.Named
 }
 
+type compatTagsURLKey struct{}
+
+func (compatTagsURLKey) String() string {
+	return "tags.url.compat"
+}
+
 // All returns all tags
 func (t *tags) All(ctx context.Context) ([]string, error) {
 	var tags []string
-
-	u, err := t.ub.BuildTagsURL(t.name)
+	var u string
+	var err error
+	var compatURL bool
+
+	// This is needed to be compatible with old registries that use "list" instead of "_list"
+	_, compatURL = ctx.Value(compatTagsURLKey{}).(bool)
```

Comment:
Instead of using context, you can just define a `func (t *tags) all(ctx context.Context, fallback bool) ([]string, error)` method which `All` calls and will call itself recursively. Passing arguments through context should be avoided whenever possible.